crates/oxc_linter/src/rules/unicorn/no_process_exit.rs

```diff
@@ -0,0 +1,183 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{ast_util::is_method_call, context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-unicorn(no-process-exit): Disallow `process.exit()`.")]
+#[diagnostic(
+    severity(warning),
+    help("Only use `process.exit()` in CLI apps. Throw an error instead.")
+)]
+struct NoProcessExitDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoProcessExit;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Disallow `process.exit()`.
+    ///
+    /// ### Why is this bad?
+    /// Only use `process.exit()` in CLI apps. Throw an error instead.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // Bad
+    /// if (problem) process.exit(1);
+    ///
+    /// // Good
+    /// if (problem) throw new Error("message");
+    ///
+    /// #! /usr/bin/env node
+    /// if (problem) process.exit(1);
+    /// ```
+    NoProcessExit,
+    correctness
+);
+
+impl Rule for NoProcessExit {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        // Allowed in shebang
+        if ctx.source_text().starts_with("#!") {
```

Comment:
does it have to be the first char?

i think it'd be better to find the conditions. then check this

```rs
        let Some(root) = ctx.nodes().iter().next() else { return };
        let AstKind::Program(program) = root.kind() else { return }
        if program.hashbang.is_some() {
            // early return
        }
```